src/components/AmplitudeAnalytics/TraceEvent.tsx

```diff
@@ -0,0 +1,71 @@
+import { Children, cloneElement, isValidElement, memo, PropsWithChildren, SyntheticEvent } from 'react'
+
+import { sendAnalyticsEvent } from '.'
+import { EventName, PartialActionNames } from './constants'
+import { ITraceContext, Trace, TraceContext } from './Trace'
+
+type TraceEventProps = {
+  actionNames: PartialActionNames
```

Comment:
This is confusing, because `action`s are actually DOM `events`. Can you rename these?

    type TraceEventProps = {
      events: Event[],
      name: EventName,
      properties?: Record<string, unknown>
    } & ITraceContext

Then change the name and type of Actions (you never use the full entry, so you can just keep the event handler name:

    export enum Event = {
      onClick = 'onClick',
    }

The example should then become simpler:

     * @example
     *  <TraceEvent events={[Event.onClick]} element={ElementName.SWAP_BUTTON}>
     *    <Button onClick={() => console.log('clicked')}>Click me</Button>
     *  </TraceEvent>

Reply:
sgtm done